### PR TITLE
995 faculty emailing

### LINKED ISSUE #995: Refactor faculty emailing on Admin Course Management page
Currently, admin can email faculty members with an `Email All` button above the faculty table. The `Email All` button should be replaced by two new buttons, `Email` and `Select All`. 

We want to add check-boxes for each faculty member that admin check to indicate which faculty members they would like to send an email to. When the `Email`  button is pressed we should populate a `mailto` with all of the selected faculty members.

### PR DESCRIPTION
**Pull request by michelmonc**
Resolves #995 
Did:
- Created a select all and email button
-Created checkboxes beside the names 
-Linked those checkboxes to the select all and email button
-Changed the class of the data tables filter from float start to float none in order to move the search button to the right

### CODE REVIEW

**Comment by doucoureb** [comment]
[Refactor faculty emailing on Admin Course Management page #995](https://trello.com/c/oDF7MfnA/660-refactor-faculty-emailing-on-admin-course-management-page-995)

File: tests/code/test_fileHandler.py [hunk 1]
```diff
@@ -47,12 +47,7 @@ def test_makingdirectory():
     eventFileStorage= [FileStorage(filename= "eventfile.pdf")]
     handledEventAttachment = FileHandler(eventFileStorage, eventId=90)
     handledEventAttachment.makeDirectory()
-    handledEventAttachment.makeDirectory()
-    assert os.path.exists('app/static/files/eventattachments/90') == True
-    # Deleting the directory
-    os.rmdir('app/static/files/eventattachments/90')
```
Comment (BrianRamsay) [inline tests/code/test_fileHandler.py]: We don't want to delete this code

File: app/static/js/slcManagement.js [hunk 2]
```diff
@@ -25,8 +25,36 @@ $(document).ready(function() {
...
+      .join(";");
+
+    if (selectedEmails) {
+      window.location.href = `mailto:${selectedEmails}?subject=Renew Course Proposal`;
+    } else {
+      alert("Please select at least one instructor to email.");
+    }
+  });
 
+  selectAllOthersButton.on("click", function () {
```
Comment (BrianRamsay) [inline app/static/js/slcManagement.js]: We should toggle the name of the Select All button if the behavior is switching

**Review by BrianRamsay — CHANGES_REQUESTED** [review]

**Review by BrianRamsay — CHANGES_REQUESTED** [review]
- you included your phone scanner code in here somehow. It shouldn't be here.
- The mailto link should open in a new window or tab. 
- The buttons should stay in the same row as the search box, like the button was before. This involved some messing with css.

File: app/static/js/slcManagement.js [hunk 3]
```diff
@@ -25,8 +25,50 @@ $(document).ready(function() {
...
+  
+      windowRef.focus();
+    
+      setTimeout(function(){
+        if(!windowRef.document) {
+            windowRef.close();
+        }
+      }, 500);
+    } else {
+      alert("Please select at least one instructor to email.");
```
Comment (AndersonStettner) [inline app/static/js/slcManagement.js]: Use flash messages not alert.

File: app/static/js/slcManagement.js [hunk 4]
```diff
@@ -25,8 +25,50 @@ $(document).ready(function() {
...
-});
+  const emailSelectedButton = $("#emailSelectedButton");
+  const selectAllOthersButton = $("#selectAllOthersButton");
+  const instructorCheckboxes = $(".instructorCheckbox");
+
+  emailSelectedButton.on("click", function () {
+    const selectedEmails = Array.from(instructorCheckboxes)
+      .filter((checkbox) => checkbox.checked)
+      .map((checkbox) => checkbox.getAttribute("data-email"))
+      .join(";");
```
Comment (AndersonStettner) [inline app/static/js/slcManagement.js]: line filter, map, and join up under from

File: app/templates/main/manageServiceLearningFaculty.html [hunk 5]
```diff
@@ -141,7 +141,41 @@ <h5 class="ms-3">There are no approved courses for {{ term.description }}.</h5>
...
         </div>
       </div>
-
+    <div id="wrapper">
+      <h1 class="text-center pt-1 pb-4">Service-Learning Faculty</h1>
+      <div class="btn" style=" text-align:right; position:relative; top: 38px; z-index:1;">
+        {% set instructorEmails = [] %}
+        {% for instructor in courseInstructors %}
+          {{ instructorEmails.append(instructor.email) or "" }}
+        {% endfor %}
```
Comment (AndersonStettner) [inline app/templates/main/manageServiceLearningFaculty.html]: You should pass the list in from the controller, do not make it with jinja.

**Review by AndersonStettner — CHANGES_REQUESTED** [review]

File: app/static/js/slcManagement.js [hunk 6]
```diff
@@ -25,8 +25,50 @@ $(document).ready(function() {
...
+            windowRef.close();
+        }
+      }, 500);
+    } else {
+      alert("Please select at least one instructor to email.");
+    }
+  });
 
+  selectAllOthersButton.on("click", function () {
+    var myButton = document.getElementById('selectAllOthersButton');
```
Comment (zachneill) [inline app/static/js/slcManagement.js]: this isn't 1845 why are you using var the only time people do is if they're stealing code from a 2007 stack overflow post

File: app/static/js/slcManagement.js [hunk 7]
```diff
@@ -25,8 +25,50 @@ $(document).ready(function() {
...
+    let uncheckedBoxFound = false;
+    for (let checkboxNum = 0; checkboxNum < instructorCheckboxes.length; checkboxNum++){
+      if (instructorCheckboxes[checkboxNum].checked == false){
```
Comment (zachneill) [inline app/static/js/slcManagement.js]: put the exclamation point somewhere instead of `== false`

File: tests/code/test_fileHandler.py [hunk 8]
```diff
@@ -41,11 +41,15 @@ def test_makingdirectory():
...
+       os.rmdir('app/static/files/eventattachments/90')
+    except OSError as e:
+        if e.errno != 2:
+            raise e    
     assert os.path.exists(os.path.join(path, str(event_id))) == False
-    
     # Creating directory and making sure it exist
     eventFileStorage= [FileStorage(filename= "eventfile.pdf")]
-    handledEventAttachment = FileHandler(eventFileStorage, eventId=90)
+    handledEventAttachment = FileHandler(eventFileStorage, eventId= 90)
```
Comment (zachneill) [inline tests/code/test_fileHandler.py:52]: ah ah ah take the space back out

File: app/static/js/slcManagement.js (hunk 4 above)
Comment (zachneill) [reply app/static/js/slcManagement.js]: snap, crackle, pop, hook line and sinker lights camera action

**Review by zachneill — DISMISSED** [review]
is the ac still down in the programmers space?

File: app/static/js/slcManagement.js (hunk 6 above)
Comment (BrianRamsay) [reply app/static/js/slcManagement.js]: The real comment here is that you shouldn't use `document.getElementById`, use the jquery methods

**Review by BrianRamsay — CHANGES_REQUESTED** [review]

**Review by BrianRamsay — APPROVED** [review]

**Pull request merged**